nncf/tensorflow/tensor_statistics/collectors.py

```diff
@@ -0,0 +1,148 @@
+ limitations under the License.
+"""
+
+from typing import Union, Deque, List
+
+import numpy as np
+import tensorflow as tf
+
+from nncf.common.tensor import NNCFTensor
+from nncf.common.tensor_statistics.collectors import MedianMADStatisticCollector
+from nncf.common.tensor_statistics.collectors import NNCFCollectorTensorProcessor
+from nncf.common.tensor_statistics.collectors import PercentileStatisticCollector
+from nncf.common.tensor_statistics.collectors import MeanPercentileStatisticCollector
+from nncf.common.tensor_statistics.collectors import MixedMinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import MeanMinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import MinMaxStatisticCollector
+from nncf.common.tensor_statistics.reduction import np_percentile_reduce_like
+from nncf.tensorflow.tensor_statistics.statistics import TFMinMaxTensorStatistic
+from nncf.tensorflow.tensor_statistics.statistics import TFPercentileTensorStatistic
+from nncf.tensorflow.tensor_statistics.statistics import TFMedianMADTensorStatistic
+from nncf.tensorflow.tensor_statistics.reduction import convert_rs_to_pt_type
+from nncf.tensorflow.tensor import TFNNCFTensor
+
+
+class TFNNCFCollectorTensorProcessor(NNCFCollectorTensorProcessor):
+    """
+    A realization of the processing methods set for TFNNCFTensors.
+    """
+
+    @staticmethod
+    def reduce_min(x: NNCFTensor, axis: Union[int, tuple, list]) -> NNCFTensor:
+        return TFNNCFTensor(tf.reduce_min(x.tensor, axis=axis))
+
+    @staticmethod
+    def reduce_max(x: NNCFTensor, axis: Union[int, tuple, list]) -> NNCFTensor:
+        return TFNNCFTensor(tf.reduce_max(x.tensor, axis=axis))
+
+    @staticmethod
+    def abs(x: NNCFTensor) -> NNCFTensor:
+        return TFNNCFTensor(tf.math.abs(x.tensor))
+
+    @staticmethod
+    def min(x1: tf.Tensor, x2: tf.Tensor) -> NNCFTensor:
+        return TFNNCFTensor(tf.math.minimum(x1.tensor, x2.tensor))
+
+    @staticmethod
+    def max(x1: tf.Tensor, x2: tf.Tensor) -> NNCFTensor:
+        return TFNNCFTensor(tf.math.maximum(x1.tensor, x2.tensor))
+
+    @staticmethod
+    def mean(x: NNCFTensor, axis: Union[int, tuple, list]) -> NNCFTensor:
+        return TFNNCFTensor(tf.math.reduce_mean(x.tensor, axis=axis))
+
+    @staticmethod
+    def stack(x: Union[List[tf.Tensor], Deque[tf.Tensor]], axis: int = 0) -> NNCFTensor:
+        x = [t.tensor for t in x]
+        return TFNNCFTensor(tf.stack(x, axis=axis))
+
+    @staticmethod
+    def unstack(x: NNCFTensor, axis: int = 0) -> List[NNCFTensor]:
+        tensor_list = tf.unstack(x.tensor, axis=axis)
+        return [TFNNCFTensor(t) for t in tensor_list]
+
+
+class TFMinMaxStatisticCollector(MinMaxStatisticCollector):
+    @staticmethod
+    def _get_processor() -> NNCFCollectorTensorProcessor:
+        return TFNNCFCollectorTensorProcessor()
+
+    def _register_input(self, x: tf.Tensor):
+        self._register_input_common(TFNNCFTensor(x))
```

Comment:
Can we unify this somehow (`TFMinMaxStatisticCollector`, `TFMixedMinMaxStatisticCollector`, `TFMeanMinMaxStatisticCollector`)? @vshampor @alexsu52

Reply:
Not worth it IMO - would have to involve adding another base class or multiple inheritance.